swift/src/Ice/AdminFacetFactory.swift

```diff
@@ -50,40 +49,39 @@ class AdminFacetFacade: ICEBlobjectFacade {
             mode: OperationMode(rawValue: mode)!,
             ctx: context,
             requestId: requestId,
-            encoding: EncodingVersion(major: encodingMajor, minor: encodingMinor))
-
-        let incoming = Incoming(
-            istr: InputStream(
-                communicator: communicator,
-                encoding: EncodingVersion(
-                    major: encodingMajor,
-                    minor: encodingMinor),
-                bytes: Data(
-                    bytesNoCopy: inEncapsBytes, count: inEncapsCount,
-                    deallocator: .none)),
-            sendResponse: sendResponse,
-            current: current)
-
-        // Dispatch directly to the servant. Do not call invoke on Incoming
-        do {
-            // Request was dispatched asynchronously if promise is non-nil
-            if let promise = try disp.dispatch(request: incoming, current: current) {
-                // Use the thread which fulfilled the promise (on: nil)
-                promise.done(on: nil) { ostr in
-                    incoming.setResult(ostr)
-                    incoming.response()
-                }.catch(on: nil) { error in
-                    incoming.exception(error)
-                }
-            } else {
-                incoming.response()
+            encoding: encoding)
+
+        let istr = InputStream(
+            communicator: communicator,
+            encoding: encoding,
+            bytes: Data(bytesNoCopy: inEncapsBytes, count: inEncapsCount, deallocator: .none))
+
+        let request = IncomingRequest(current: current, inputStream: istr)
+
+        // Dispatch directly to the servant.
+        dispatcher.dispatch(request).map { response in
```

Comment:
This is largely the same code as in ObjectAdapterI. It would be nice to consolidate it in a follow-up PR.